### PR TITLE
Exclude failed runs from summary table

### LINKED ISSUE #49: Exclude failed runs from the summary's numbers
**Description of the problem / feature request:**

This is a follow-up of #39. We'd like to exclude failed runs (exit code != 0) from the final numbers displayed in the summary tables.

**Feature requests: what underlying problem are you trying to solve with this feature?**

Failed runs aren't very meaningful in terms of performance benchmarking.

### PR DESCRIPTION
**Pull request by joeleba**
**What this PR does and why we need it:**

This PR excludes failed runs from the summary table and also clearly state which run failed with which exit code.

Screenshot:
Screenshot from 2019-07-01 16-41-29

**New changes / Issues that this PR fixes:**

Fixes #49.

### CODE REVIEW

File: benchmark.py [hunk 1]
```diff
@@ -415,35 +416,51 @@ def print_summary(data):
...
     last_collected = collected
-    if list(filter(lambda x: x != 0, exit_statuses)):
+    if non_zero_runs:
       print(
-          ('The runs contain non-zero exit code(s): %s. Please check the full '
-           'log for more details.' % exit_statuses))
+          ('The following runs contain non-zero exit code(s):\n %s\n'
+           'Please check the full log for more details. These runs are '
+           'excluded from the above result table.' % '\n '.join(
+               '- run: %s/%s, exit_code: %s' % (k + 1, num_runs, v)
```
Comment (meisterT) [inline benchmark.py]: I don't think we need the total number of runs repeated here

File: benchmark.py [hunk 2]
```diff
@@ -415,35 +416,51 @@ def print_summary(data):
...
-              ('% 7.3fs' % values.stddev()).center(10),
+              ('% 8.3fs %s' % (
+                  values_exclude_failures.mean(), mean_diff)).center(20),
+              ('% 8.3fs %s' % (
+                  values_exclude_failures.median(), median_diff)).center(20),
+              ('% 7.3fs' % values_exclude_failures.stddev()).center(10),
               pval.center(10)))
     last_collected = collected
-    if list(filter(lambda x: x != 0, exit_statuses)):
+    if non_zero_runs:
```
Comment (meisterT) [inline benchmark.py]: have you considered grouping them by exit code?

File: benchmark.py (hunk 1 above)
Comment (joeleba) [reply benchmark.py]: I wanted to be consistent with what we print during the runs. E.g.:
```
I0701 16:40:14.391463 140566465873728 logger.py:34] Starting benchmark run 2/4:
I0701 16:40:14.392107 140566465873728 logger.py:34] Executing Bazel command: bazel build --nostamp --noshow_progress --color=no --nobuild //src:bazel
```

File: benchmark.py (hunk 2 above)
Comment (joeleba) [reply benchmark.py]: I actually didn't. I feel that displaying the non-zero runs one by one would make the log more readable & easier to comprehend. What would you suggest as a group-by-exit-code message? Something like this?
```
The following runs contain non-zero exit code(s):
 - exit_code=1: runs 1/4, 2/4
 - exit_code=2: runs 3/4
```

**Pull request merged**